refactor(testing): improve layout and naming of `testing` packages

danceratopz opened the pull request:
## 🗒️ Description

The weld provides us with a one-off opportunity to break everything. This PR improves the naming of the EEST libraries that have been added to `./packages/test/src`.

Note: We've moved from `test` to `testing` to better express the semantics of the package. The spec `./tests` were part of the EEST repo, but are not part of this package, which just contains the helper libraries, test generation framework, test running frameworks (`consume` and `execute`) and other cli helpers (e.g., `gentest`).

- Moves `packages/test` to `packages/testing`.
- Adds a top-level dir/package `packages/testing/src/execution_testing/`.
- Improves the `imports` by providing commonly used objects on `execution_testing`'s top-level.
- Renames sub-packages as below.
```
ethereum_test_base_types  →  execution_testing.base_types  
ethereum_test_benchmark   →  execution_testing.benchmark  
ethereum_test_checklists  →  execution_testing.checklists  
ethereum_clis             →  execution_testing.client_clis  
config                    →  execution_testing.config  
cli                       →  execution_testing.cli  
pytest_plugins.<anything> →  execution_testing.cli.pytest_commands.plugins.<anything>  
ethereum_test_exceptions  →  execution_testing.exceptions  
ethereum_test_execution   →  execution_testing.execution  
ethereum_test_fixtures    →  execution_testing.fixtures  
ethereum_test_forks       →  execution_testing.forks  
ethereum_test_rpc         →  execution_testing.rpc  
ethereum_test_specs       →  execution_testing.specs  
ethereum_test_tools       →  execution_testing.tools  
ethereum_test_types       →  execution_testing.test_types  
ethereum_test_vm          →  execution_testing.vm  
```

### Modifications to the original Plan in the Issue

@kclowes, here's a couple of modifications I made to https://github.com/ethereum/execution-spec-tests/issues/2209#issuecomment-3340578295

0. `tests` -> `testing` (there are no tests in the package).
1. For `ethereum_test_types`, we decided on `types`, but I used `test_types` to prevent shadowing Python's builtin `types` module:
    ```
    ethereum_test_types       →  execution_testing.test_types
    ```
2. `ethereum_test_cli` -> `execution_tests.cli`.
3. We talked about nesting `pytest_plugins` directly under `ethereum_test_cli` (which is now also just `cli`), but that has a `pytest_commands` sub-directory, so I moved it underneath there, but as `plugins` (after all!).
    ```
    pytest_plugins.<anything> →  execution_tests.cli.pytest_commands.plugins.<anything>
    ```

New modules since the renaming concept was created in https://github.com/ethereum/execution-spec-tests/issues/2209:
```
# in packages/tests/
git mv src/ethereum_test_benchmark/ src/execution_testing/benchmark
git mv src/ethereum_test_checklist src/execution_testing/checklist
```
## 🔗 Related Issues or PRs
Closes https://github.com/ethereum/execution-spec-tests/issues/2209 :
 - https://github.com/ethereum/execution-spec-tests/issues/2209


## ✅ Checklist
<!-- Please check off all required items. For those that don't apply remove them accordingly. -->

- [ ] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx --with=tox-uv tox -e static
    ```
- [ ] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [ ] All: Considered adding an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.
- [ ] Tests: For PRs implementing a missed test case, update the [post-mortem document](/ethereum/execution-spec-tests/blob/main/docs/writing_tests/post_mortems.md) to add an entry the list.
- [ ] Ported Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) or [tests/static](/ethereum/execution-spec-tests/blob/main/tests/static) have been assigned `@ported_from` marker.

#### Cute Animal Picture

<img width="512" alt="image" src="https://github.com/user-attachments/assets/a48e7d6c-850b-4172-b6e6-7c4239898d23" />

danceratopz committed:
chore(tests): create top-level package; rename subpackages

danceratopz committed:
chore(tests): add top-level ini file

danceratopz committed:
chore(tests): update packages and paths in pyproject.toml

danceratopz committed:
chore(tests): update plugin paths used in top-level conftests

danceratopz committed:
refactor(tests): update imports for base_types, exceptions and vm

danceratopz committed:
refactor(tests): update imports for specs, fixtures and forks

danceratopz committed:
refactor(tests): update imports for tools, types and execution

danceratopz committed:
refactor(tests): update config, client and remaining imports

danceratopz committed:
refactor(tests): update string path and package references

danceratopz committed:
refactor(tests): fix remaining imports for mypy

danceratopz committed:
chore(tests): ignore type for `TransactionGeneric`'s `chain_id` default

danceratopz committed:
chore(tests): single py.typed at top package level; add stubs to config

danceratopz committed:
chore(tests): ruff format

danceratopz committed:
refactor(tests): update pytest ini files to reflect package renames

danceratopz committed:
refactor(tests): update imports for base_types/exceptions/vm in tests

danceratopz committed:
refactor(tests): update imports for tools/types/execution in tests

danceratopz committed:
refactor(tests): update remaining imports in ./tests/

danceratopz committed:
chore(tests): ruff format tests

danceratopz committed:
chore(tests): organize formats

marioevz commented:
- `ethereum_execution_tests` -> `eest`
- `packages/tests` -> `packages/eest`
- Update pyproject.toml

kclowes committed:
chore(tests): execution_spec_tests -> eest

kclowes committed:
chore(tests): Appease ruff

kclowes committed:
chore(tests): Fix tests imports

kclowes committed:
chore(tests): Remove some warnings

marioevz committed:
refactor(packages/tests): Refactor eest.tools and package exports

marioevz committed:
refactor(tests): Refactor eest.tools imports

kclowes commented on packages/testing/src/execution_testing/__init__.py:
note to self: this isn't right. edit: Actually, I don't think we need this py.typed since we have one up a level in the eest dir.

kclowes commented on packages/testing/src/eest/client_clis/cli_types.py:
```diff
@@ -6,31 +6,35 @@
 from pydantic import Field, PlainSerializer, PlainValidator
 
-from ethereum_test_base_types import (
+from eest.base_types import (
```
This is much better, nice!

kclowes commented on packages/tests/src/eest/ethereum_test_cli/pytest_commands/plugins/execute/pre_alloc.py:
```diff
@@ -9,35 +9,34 @@
-from ethereum_test_types import ChainConfig, TransactionTestMetadata
-from ethereum_test_types.eof.v1 import Container
-from ethereum_test_vm import Bytecode, EVMCodeType, Opcodes
+from eest.test_types import Alloc as BaseAlloc
```
This shouldn't block merging at all, but I'm curious why we import these `as` lines separately instead of putting it with the rest of the `from eest.test_types` imports?

kclowes commented on packages/testing/src/eest/ethereum_test_cli/pytest_commands/plugins/execute/rpc/hive.py:
```diff
@@ -13,19 +13,17 @@
+from eest.rpc import EngineRPC, EthRPC
+from eest.base_types import Account, Hash
+from eest.test_types import EOA, Alloc, Environment, Withdrawal
+from eest.test_types import ChainConfig, Requests
```
nit: These can be combined

kclowes commented on packages/testing/src/eest/ethereum_test_cli/pytest_commands/plugins/filler/tests/test_filler.py:
```diff
@@ -11,8 +11,11 @@
 
-from ethereum_test_tools import Environment
-from ethereum_clis import ExecutionSpecsTransitionTool, TransitionTool
+from eest.test_types import Environment
```
nit:
```suggestion
from eest import Environment
```
or alternatively, the docstring lines below should read `from eest.test_types import Environment`

kclowes reviewed (commented):
I got about halfway through today. I noticed a few places where import lines could be combined, and some minor inconsistencies in how we're importing things from either `eest` or `eest.test_types`, but nothing major so far! I'll pick up reviewing after we've decided on which naming scheme to go with!

marioevz committed:
refactor(packages): Rename `packages/tests` -> `packages/testing`

marioevz committed:
fix(gitignore): Fix `fixtures` ignore

marioevz committed:
refactor(packages): Rename module `eest` -> `execution_testing`

marioevz committed:
refactor(packages): Rename package `eest` -> `execution-testing`

danceratopz committed:
fix(testing): update paths used by the jinja2 template loader

danceratopz committed:
fix(testing): fix location of `env.yaml`

danceratopz committed:
chore(testing): rename simulator prefix from eest to eels

For consistency with the Hive simulator naming, cf ethereum/hive#1362.

danceratopz committed:
chore(testing): rename package to ethereum-execution-testing

kclowes committed:
refactor(testing): ethereum_test_cli -> cli

SamWilsn commented on .gitignore:
```diff
@@ -75,7 +75,10 @@ tests/execution-spec-generated-tests
 *.log
+
+# config (output of `uv run make env`)
+env.yaml
```
Avoid removing the trailing newline.

danceratopz committed:
fix(testing): remove unimported `vm` from `__all__`

It's not imported in the top-level init.py file;
doing an `import vm from ..` will cause an attribute error.
This keeps things inline with other sub-packages.

danceratopz committed:
chore(testing): update `eof.py` docstrings to use `execution_testing.*`

Update docstring references from `ethereum_test_*` to `execution_testing.*`
for consistency with the package rename.

danceratopz committed:
fix(testing): update docstrings to use `execution_testing.*` (specs)

  Update module references from `ethereum_test_*` to `execution_testing.*`
  in test docstrings and imports. Also fixes import in skipped checklist test
  for when it's re-enabled.

danceratopz committed:
fix(testing): remove `eels_resolutions.json` from package data

danceratopz committed:
feat(testing): add `Op` as ergonomic alias for `Opcodes`

Export `Op` from `execution_testing` package root to simplify imports.
Users can now write
`from execution_testing import Op`
instead of
`from execution_testing import Opcodes as Op`.

danceratopz commented on packages/testing/src/execution_testing/config/env.py:
```diff
@@ -26,7 +26,18 @@
 
-ENV_PATH = Path(__file__).resolve().parent.parent.parent / "env.yaml"
+
+def get_project_root() -> Path:
+    """Find project root by locating .git directory."""
+    current = Path(__file__).resolve().parent
+    while current != current.parent:
+        if (current / ".git").exists():
+            return current
+        current = current.parent
+    raise FileNotFoundError("Could not find project root (no .git directory)")
```
This is a good catch. Follow-up issue:
- https://github.com/ethereum/execution-specs/issues/1674

danceratopz commented on packages/testing/pyproject.toml:
```diff
@@ -6,9 +6,9 @@ requires = [
+name = "ethereum-execution-testing"
 version = "1.0.0"
-description = "Ethereum execution client test authoring framework"
+description = "Ethereum execution layer client test generation and runner framework"
```
It's ok. Can be improved in a follow-up issue.

danceratopz commented on packages/testing/src/execution_testing/cli/gentest/source_code_generator.py:
```diff
@@ -82,7 +82,7 @@ def format_code(code: str) -> str:
             formatter_path = Path(sys.prefix) / "bin" / "ruff"
 
         # Call ruff to format the file
```
`uv run eest make test` runs correctly, but the generated test module is not formatted. Follow-up issue:
- https://github.com/ethereum/execution-specs/issues/1675

danceratopz commented on packages/testing/src/execution_testing/cli/fuzzer_bridge/performance_utils.py:
```diff
@@ -105,9 +105,9 @@ def calculate_optimal_workers(file_count: int) -> int:
     try:
-        import ethereum_test_forks
+        from execution_testing.forks import get_fork_by_name
 
-        return getattr(ethereum_test_forks, fork_name, None)
+        return get_fork_by_name(fork_name)
```
This one looks ok to me; it returns `None` if no such fork is found; no exception is raised. I would prefer an exception tbh, but I don't have the full context, it's fine for now.

https://github.com/danceratopz/execution-specs/blob/e7874d9007df2c662bfbbf6f5be0558f4526d882/packages/testing/src/execution_testing/forks/helpers.py#L323-L328

spencer-tb replied:
Yeah think its fine too.

danceratopz commented on packages/testing/src/execution_testing/cli/fuzzer_bridge/models.py:
```diff
@@ -36,17 +36,14 @@ class FuzzerAccountInput(BaseModel):
     EEST's Account validation logic or defaults.
     """
 
+    model_config = ConfigDict(populate_by_name=True)
+
     balance: HexNumber
     nonce: HexNumber = HexNumber(0)
     code: Bytes = Bytes(b"")
     storage: Dict[HexNumber, HexNumber] = Field(default_factory=dict)
     private_key: Hash | None = Field(None, alias="privateKey")
```
This was added in this commit https://github.com/ethereum/execution-specs/commit/829493eb202ea54ec9482c605a514c6da0062f7e

Looks ok, but tagging @kclowes to double-check :pray:

danceratopz reviewed (commented):
This would be good to go from my side! I did spot one remaining warning from `fill` but I'm not sure if it was there before or not: 

Command:
```
uv run fill tests/istanbul/eip1344_chainid/test_chainid.py::test_chainid --output=/tmp/fixtures --clean --fork=Osaka
```

Warning:
```
=============================== warnings summary ===============================
.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:812
  /home/dtopz/code/github/new/execution-specs/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py:812: PytestAssertRewriteWarning: Module already imported so cannot be rewritten; execution_testing.cli.pytest_commands.plugins.custom_logging.plugin_logging
    self.import_plugin(arg, consider_entry_points=True)
```

Made a follow-up issue for this:
- https://github.com/ethereum/execution-specs/issues/1676
I don't mind if it's solved here or in a follow-up PR.

spencer-tb reviewed (approved):
LGTM! Thanks guys. Fixture hashes are the same so filling all good. Consume seemed to work fine too. Feel free to merge from my side.

kclowes commented:
I think a follow up PR makes sense so we can get everyone unblocked here! Happy to take up #1676 today.

marioevz committed:
refactor(testing): Usages of Opcodes

marioevz reviewed (approved):
Amazing work @danceratopz and @kclowes! This looks great to me, I feel like it's ready to merge. I pushed one last commit at the end to make a tiny change.

marioevz committed:
fix(uv.lock): `ethereum-execution-testing`

fselmo reviewed (approved):
lgtm! Great work 🎉

marioevz committed:
fix(testing): Fix consume commands
